src/plugins/workspace/public/components/workspace_detail/workspace_detail.tsx

```diff
@@ -123,6 +134,48 @@ export const WorkspaceDetail = (props: WorkspaceDetailProps) => {
     formData.selectedDataSourceConnections,
   ]);
 
+  // When user has unsaved changes and navigates to other page, will show a confirm modal.
+  useEffect(() => {
+    onAppLeave((actions) => {
+      if (!props.isFormSubmitting && isEditing && numberOfChanges > 0) {
+        return actions.confirm(
+          i18n.translate('workspace.detail.navigate.message', {
+            defaultMessage: 'Any unsaved changes will be lost.',
+          }),
+          i18n.translate('workspace.detail.navigate.title', {
+            defaultMessage: 'Navigate away?',
+          }),
+          handleResetForm
```

Comment:
So no matter user clicks `cancel` or `confirm`, the form data will be reset?

Reply:
No, the form data will be reset only when users click confirm button.

Reply:
In this line (https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8317/files#diff-94b600f79770d5081137b8938222dfc2718295a199b0685cd3fea4040114f451R520) when user clicks 'cancel' button, the callback function will dose not execute.